shared/src/pool_fetching.rs

```diff
@@ -243,18 +243,18 @@ struct FetchedPool {
     token1_balance: Result<U256, MethodError>,
 }
 
-fn handle_results(results: Vec<FetchedPool>) -> Result<Vec<Pool>> {
-    // Node errors should be bubbled up but contract errors should lead to the pool being skipped.
-    fn handle_contract_error<T>(result: Result<T, MethodError>) -> Result<Option<T>> {
-        match result {
-            Ok(t) => Ok(Some(t)),
-            Err(err) => match EthcontractErrorType::classify(&err) {
-                EthcontractErrorType::Node => Err(err.into()),
-                EthcontractErrorType::Contract => Ok(None),
-            },
-        }
+// Node errors should be bubbled up but contract errors should lead to the pool being skipped.
+pub fn handle_contract_error<T>(result: Result<T, MethodError>) -> Result<Option<T>> {
+    match result {
+        Ok(t) => Ok(Some(t)),
+        Err(err) => match EthcontractErrorType::classify(&err) {
+            EthcontractErrorType::Node => Err(err.into()),
+            EthcontractErrorType::Contract => Ok(None),
+        },
     }
+}
```

Comment:
Moving this out of `handle_results` and making public to be used in balancer pool fetching.